x-pack/solutions/security/plugins/security_solution/public/agent_builder/helpers.tsx

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ESSENTIAL_ALERT_FIELDS } from '../../common';
+
+/**
+ * Filters raw alert data to only include essential fields and stringifies the result.
+ * This reduces context window usage by keeping only the most relevant information.
+ */
+export const filterAndStringifyAlertData = (rawData: Record<string, string[]>): string => {
```

Comment:
Could we make it more readable?

My suggestions
- rename the function to `stringifyEssentialAlertData`
- simplify the implementation for readability. For example Lodash's `pick` should do the same job but is one liner